packages/pigeon/lib/dart_generator.dart

```diff
@@ -434,6 +452,194 @@ final BinaryMessenger? ${_varNamePrefix}binaryMessenger;
+          codecName: codecName,
+          codecInstanceName: codecInstanceName,
+          superClassApi: api.superClass?.associatedProxyApi,
+          unattachedFields: api.unattachedFields,
+          flutterMethodsFromSuperClasses: flutterMethodsFromSuperClasses,
+          flutterMethodsFromInterfaces: flutterMethodsFromInterfaces,
+          declaredFlutterMethods: api.flutterMethods,
+        ))
+        ..constructors.add(
+          _proxyApiDetachedConstructor(
+            apiName: api.name,
+            superClassApi: api.superClass?.associatedProxyApi,
+            unattachedFields: api.unattachedFields,
+            flutterMethodsFromSuperClasses: flutterMethodsFromSuperClasses,
+            flutterMethodsFromInterfaces: flutterMethodsFromInterfaces,
+            declaredFlutterMethods: api.flutterMethods,
+          ),
+        )
+        ..fields.addAll(<cb.Field>[
+          if (api.constructors.isNotEmpty ||
+              api.attachedFields.any((ApiField field) => !field.isStatic) ||
+              api.hostMethods.isNotEmpty)
+            _proxyApiCodecInstanceField(
+              codecInstanceName: codecInstanceName,
+              codecName: codecName,
+            ),
+        ])
+        ..fields.addAll(_proxyApiUnattachedFields(api.unattachedFields))
+        ..fields.addAll(_proxyApiFlutterMethodFields(
+          api.flutterMethods,
+          apiName: api.name,
+        ))
+        ..fields.addAll(_proxyApiInterfaceApiFields(apisOfInterfaces))
+        ..fields.addAll(_proxyApiAttachedFields(api.attachedFields))
+        ..methods.add(
+          _proxyApiSetUpMessageHandlerMethod(
+            flutterMethods: api.flutterMethods,
+            apiName: api.name,
+            dartPackageName: dartPackageName,
+            codecName: codecName,
+            unattachedFields: api.unattachedFields,
+            hasCallbackConstructor: api.flutterMethods
+                .followedBy(flutterMethodsFromSuperClasses)
+                .followedBy(flutterMethodsFromInterfaces)
+                .every((Method method) => !method.isRequired),
+          ),
+        )
+        ..methods.addAll(
+          _proxyApiAttachedFieldMethods(
+            api.attachedFields,
+            apiName: api.name,
+            dartPackageName: dartPackageName,
+            codecInstanceName: codecInstanceName,
+            codecName: codecName,
+          ),
+        )
+        ..methods.addAll(_proxyApiHostMethods(
+          api.hostMethods,
+          apiName: api.name,
+          dartPackageName: dartPackageName,
+          codecInstanceName: codecInstanceName,
+          codecName: codecName,
+        ))
+        ..methods.add(_proxyApiCopyMethod(
+          apiName: api.name,
+          unattachedFields: api.unattachedFields,
+          declaredAndInheritedFlutterMethods: flutterMethodsFromSuperClasses
+              .followedBy(flutterMethodsFromInterfaces)
+              .followedBy(api.flutterMethods),
+        )),
+    );
```

Comment:
This seems pretty nice. Definitely something to consider for the other Dart generation.